include/rx/ranges.hpp

```diff
@@ -817,7 +818,7 @@ struct transform {
         transform transform_;
         using output_type = decltype(std::declval<const F&>()((input_.get())));
         static constexpr bool is_finite = is_finite_v<InputRange>;
-        static constexpr bool is_idempotent = is_idempotent_v<InputRange>;
+        static constexpr bool is_idempotent = false; // `func` is called each time `get` is called
```

Comment:
Maybe add a `static_assert(is_idempotent_v<InputRange>)` to the class body?

Reply:
Oh right, that wouldn't actually be right. This seems fine.

Do you think it would make sense to actually derive this value from the const-invocability of `F`, i.e. `std::is_invocable_r_v<output_type, const F&, input_type>`?

(A similar change could be made to `generate`.)

Reply:
I'm not familiar with any method to check if an invocation is pure... And `std::is_invocable_r_v<output_type, const F&, input_type>` doesn't work with lambdas capturing by reference: https://godbolt.org/z/6qjEv-

Reply:
Wow, that is pretty surprising! I would definitely expect lambdas to have a non-const `operator()` when capturing by non-const reference. Though I suppose even that would not be good enough, since they can have internally mutable state.

Anyway, looks good then!

Reply:
`mutable` drop the `const` requirement in its `operator()`

Reply:
If you want, you could add a template parameter to override that property.